core/src/wallet/common/AbstractAccount.cpp

```diff
@@ -187,16 +187,21 @@ namespace ledger {
         void AbstractAccount::emitEventsNow() {
             auto self = shared_from_this();
             run([self] () {
-                for (auto& event : self->_events) {
+                std::list<std::shared_ptr<api::Event>> events;
+                {
+                    std::lock_guard<std::mutex> lock(self->_eventsLock);
```

Comment:
This is a solution, but if the list of events is large, this is not a real solution and a lock-free solution, ring buffer or immutable slice lock (i.e. you take the size and prevent other threads to delete elements until you’re done) would be faster. However, I think it’s fine.

Reply:
Didn't get how the size of the list affect this approach, swap is constant time.